symfony/plugins/orangehrmPimPlugin/Api/ValidationEmployeeEmailAPI.php

```diff
@@ -0,0 +1,116 @@
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Pim\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Model\ArrayModel;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\ResourceEndpoint;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Pim\Traits\Service\EmployeeServiceTrait;
+
+class ValidationEmployeeEmailAPI extends Endpoint implements ResourceEndpoint
+{
+    use EmployeeServiceTrait;
+
+    public const PARAMETER_WORK_EMAIL = 'workEmail';
+    public const PARAMETER_IS_CHANGEABLE_WORK_EMAIL = 'valid';
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $empNumber = $this->getRequestParams()->getInt(RequestParams::PARAM_TYPE_ATTRIBUTE, CommonParams::PARAMETER_EMP_NUMBER);
+        $workEmail = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_WORK_EMAIL);
+        $employee = $this->getEmployeeService()->getEmployeeDao()->getEmployeeByEmpNumber($empNumber);
+        $this->throwRecordNotFoundExceptionIfNotExist($employee,Employee::class);
+        $employeeCurrentWorkEmail = $employee->getWorkEmail();
+        if ($employeeCurrentWorkEmail){
+            $isChangeableWorkEmail = $this->getEmployeeService()->getEmployeeDao()->isWorkEmailAvailableByCurrentEmail($workEmail,$employeeCurrentWorkEmail);
+        }
+        else{
+            $isChangeableWorkEmail = $this->getEmployeeService()->getEmployeeDao()->isWorkEmailAvailable($workEmail);
+        }
+        return new EndpointResourceResult(
+            ArrayModel::class,
+            [
+                self::PARAMETER_IS_CHANGEABLE_WORK_EMAIL => $isChangeableWorkEmail,
+                self::PARAMETER_WORK_EMAIL => $workEmail,
+            ]
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(self::PARAMETER_WORK_EMAIL, new Rule(Rules::STRING_TYPE),
+                new Rule(Rules::CALLBACK, [
+                function () {
+                    $workEmail = $this->getRequestParams()->getString(RequestParams::PARAM_TYPE_QUERY, self::PARAMETER_WORK_EMAIL);
+                    return filter_var($workEmail, FILTER_VALIDATE_EMAIL);
+                }
+            ])),
+            new ParamRule(CommonParams::PARAMETER_EMP_NUMBER),
```

Comment:
Add validation rule, `new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)`

Reply:
if we added this, then, if user logged in as  ESS  (not supervisors) API will not work??